### DISCUSSION TITLE
Plugin trait not satisfied

### DISCUSSION THREAD

**Discussion by DasLixou** [bevyengine/bevy#6650] [Q&A] [answered]
i get this error:
```
error[E0277]: the trait bound `BevyVfxBagPlugin: bevy::prelude::Plugin` is not satisfied
   --> src/camera.rs:22:24
    |
22  |         app.add_plugin(BevyVfxBagPlugin) // This needs to be added for any effect to work
    |             ---------- ^^^^^^^^^^^^^^^^ the trait `bevy::prelude::Plugin` is not implemented for `BevyVfxBagPlugin`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the following other types implement trait `bevy::prelude::Plugin`:
              AnimationPlugin
              AssetCountDiagnosticsPlugin<T>
              AssetPlugin
              AudioPlugin
              BloomPlugin
              CameraProjectionPlugin<T>
              ColorMaterialPlugin
              Core2dPlugin
            and 46 others
```

when using `bevy-vfx-bag` but also with the egui-inspector. i dont know what it does but its in my ide and on compile even when there is a Plugin implemented for it

**Comment by alice-i-cecile** [comment, marked as answer]
This error is generally seen when you have two different versions of Bevy: your app may implement Bevy 0.9's `Plugin` trait, while the plugin you're trying to use is using Bevy 0.8's `Plugin` trait. Definitely a poor error message.

Use `cargo tree` (or inspect your dependencies carefully) to figure out where the version mismatch is coming from.

**Reply by DasLixou** [reply]
Oh! That makes sense! i also was quite annoyed that i had to compile the latest commit (my repo) AND the official 0.9 release.. is there a way to still make it work? should i just straight copy the repo into mine and change the version? or fork it? or is there something simpler?

**Reply by mockersf** [reply]
`bevy-vfx-bag` only works with Bevy 0.9, so your repo is probably a fork of Bevy at a previous version (0.8 or 0.9.0-dev?)

You could fork bevy-vfx-bag and try to make it work with the same Bevy version you have, or update your Bevy version. If possible, I would recommend the latter.

**Reply by DasLixou** [reply]
> `bevy-vfx-bag` only works with Bevy 0.9, so your repo is probably a fork of Bevy at a previous version (0.8 or 0.9.0-dev?)
> 
> You could fork bevy-vfx-bag and try to make it work with the same Bevy version you have, or update your Bevy version. If possible, I would recommend the latter.

The problem is that im not behind. im at the newest commit of bevy. so updating mine is downgrading which isn't a good idea because i use things that were added after 0.9. so forking?

**Reply by alice-i-cecile** [reply]
Ah, in that case you'll want to follow the advice here: https://bevy-cheatbook.github.io/setup/bevy-git.html

**Reply by mockersf** [reply]
you can patch the Bevy used by the plugin by using:
```
[patch.crates-io]
bevy = { path = "path/to/your/bevy" }
```
or however you choose to depend on your own Bevy

**Reply by DasLixou** [reply]
Thanks :D
